roles/setup_gitops/tasks/main.yml

```
@@ -0,0 +1,119 @@
+---
+- name: Get available CSVs
+  kubernetes.core.k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: ClusterServiceVersion
+    namespace: "{{ sg_gitops_namespace }}"
+  register: _sg_current_csvs
+
+- name: Fail if gitops CSV is not present
+  vars:
+    csv_details: "resources[*].metadata.name"
+    csvs: "{{ _sg_current_csvs | json_query(csv_details) }}"
+  ansible.builtin.fail:
+    msg: "No CSV for openshift-gitops-operator are present"
+  when:
+    - (csvs | select('match', 'openshift-gitops-operator') | list | length) == 0
+
+- name: Mirror required images to local registry
+  ansible.builtin.include_tasks: local-registry.yml
+  when:
+    - sg_local_registry | default('') | length > 0
```

Has a default already:
```suggestion
    - sg_local_registry | length > 0
```